idaes/core/util/tests/test_model_diagnostics.py

```diff
@@ -999,6 +1014,111 @@ def test_display_near_parallel_variables(self):
     v1, v4
     v2, v4
 
+====================================================================================
+"""
+
+        assert stream.getvalue() == expected
+
+    @pytest.mark.component
+    def test_collect_constraint_mismatches(self):
+        m = ConcreteModel()
+        m.v1 = Var(initialize=2)
+        m.v2 = Var(initialize=3)
+
+        # Constraint with no free variables
+        m.c1 = Constraint(expr=m.v1 == m.v2)
+        m.v1.fix()
+        m.v2.fix()
```

Comment:
As it stands, Arc constraints will get lumped in here, however from a strict sense they should fall in here (as they can be trivially eliminated). However, from a structural stand point we probably want to ignore these.

What do we want to allow? One option would be the special case of `a=b` where `a` and `b` are leaf nodes (i.e. not expressions themselves).

Reply:
@dallan-keylogic I believe I have fixed this - the cancellation checks are only run if at least one of the sides in an (in)equality is a sum expression. This means that constraints with the form `a == b` and `a == f(b)` will not be flagged as cancelling.

Reply:
Also, regarding you initial comment, this a tool to do that could be written fairly easily as it would just iterate over a `ComponentSet` containing the `Constraint` of interest and call this walker for each `Constraint`. This walker is not part of the `DiagnosticsToolbox` class, so it can easily be called from other places too (such as the `SVDToolbox`) if we can come up with a way to identify a `set of `Constraints` of interest.